build-system/pr-check.js

```diff
@@ -126,7 +126,9 @@ function isBuildSystemFile(filePath) {
       !isFlagConfig(filePath) &&
       // Exclude visual diff files from build-system since we want it to trigger
       // visual diff tests.
-      !isVisualDiffFile(filePath);
+      !isVisualDiffFile(filePath))
+      // OWNERS.yaml files should trigger build system to run tests
+      || path.basename(filePath) === 'OWNERS.yaml';
```

Comment:
I see. The change you've made in the line above will merely make `isBuildSystemFile` return true if any of the `OWNERS.yaml` files are modified, which will result in `gulp ava` being run as part of `pr-check.js`.

I'd imagine you'll need to add a new `OWNERS` file type, a new `isOwnersFile` check, and a new `runOwnersTests` method that calls the test you linked to in order to have the level of coverage you seek.

To enforce the fact that `CODEOWNERS` is updated whenever `OWNERS.yaml` is touched, you could do something like [this](https://github.com/ampproject/amphtml/blob/master/build-system/pr-check.js#L368).